addons/stock/static/src/views/search/stock_orderpoint_search_panel.js

```diff
@@ -1,16 +1,23 @@
 /** @odoo-module **/
 
-import { useState } from '@odoo/owl';
+import { useService } from "@web/core/utils/hooks";
+import { onWillStart, useState } from '@odoo/owl';
 import { SearchPanel } from "@web/search/search_panel/search_panel";
 
 
 export class StockOrderpointSearchPanel extends SearchPanel {
     static template = "stock.StockOrderpointSearchPanel";
 
     setup() {
+        this.orm = useService("orm");
         super.setup(...arguments);
         this.globalVisibilityDays = useState({value: 0});
         this.state.sidebarExpanded = false;
+        onWillStart(this.getVisibilityParameter);
+    }
+
+    async getVisibilityParameter() {
+        this.globalVisibilityDays.value = await this.orm.call("ir.config_parameter", "get_param", ["stock.visibility_days", 0]);
```

Comment:
Be warned when the parameter exists, you get it as a string, and we it doesn't, you got 0 as a number.
Since `applyGlobalVisibilityDays` parses it to always be a int, I think it should be good to do the same here for consistence sake